src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -970,6 +971,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         const auto vkey = gsl::narrow_cast<WORD>(e.OriginalKey());
         const auto scanCode = gsl::narrow_cast<WORD>(e.KeyStatus().ScanCode);
 
+        // Short-circuit isReadOnly check to avoid warning dialog
         if (_isReadOnly)
         {
             e.Handled(!keyDown || _TryHandleKeyBinding(vkey, scanCode, modifiers));

```

Comment:
Hey I know this isn't a part of this PR, but should this block be _below_ the enhanced key and alt+# branches? Like move it to ~line 997? It seems like those two branches might affect the  behavior of the keybinding handling